src/main/java/com/vistatec/ocelot/segment/view/SegmentView.java

```diff
@@ -296,18 +310,102 @@ protected void updateRowHeights() {
+//                    segmentCell.setFont(targetFontMetrics.getFont());
+//                    segmentCell.setText(text);
+//                    int textLinesCount = countTextLines(segmentCell);
+//                    if(textLinesCount > 0){
+//	                    int cellHeight = (textLinesCount * targetFontMetrics.getHeight()) + (textLinesCount - 1)*targetFontMetrics.getLeading();
+//	                    rowHeight = Math.max(cellHeight, rowHeight);
+//                    }
+//                } else if (col == 3) {
+//                    String text;
+//                    if (enabledTargetDiff) {
+//                        List<String> textDiff = segmentTableModel.getSegment(modelRow).getTargetDiff();
+//                        StringBuilder displayText = new StringBuilder();
+//                        for (int i = 0; i < textDiff.size(); i += 2) {
+//                            displayText.append(textDiff.get(i));
+//                        }
+//                        text = displayText.toString();
+//                    } else {
+//                        text = segmentTableModel.getSegment(modelRow).getOriginalTarget().getDisplayText();
+//                    }
+//                    segmentCell.setFont(targetFontMetrics.getFont());
+//                    segmentCell.setText(text.toString());
+//                    int textLinesCount = countTextLines(segmentCell);
+//                    if(textLinesCount > 0){
+//                    	int cellHeight = (textLinesCount * targetFontMetrics.getHeight()) + (textLinesCount - 1)*targetFontMetrics.getLeading();
+//	                    rowHeight = Math.max(cellHeight, rowHeight);
+//                    }
+//                }
+                // Need to set width to force text area to calculate a pref height
+//                segmentCell.setSize(new Dimension(width, sourceTargetTable.getRowHeight(viewRow)));
+//                rowHeight = Math.max(rowHeight, segmentCell.getPreferredSize().height);
+//            }
+//>>>>>>> marta_branch:src/main/java/com/vistatec/ocelot/segment/SegmentView.java
             sourceTargetTable.setRowHeight(viewRow, rowHeight);
         }
         setViewportView(sourceTargetTable);
     }
+    
+	/**
+	 * Counts how many text lines are displayed in the text pane passed as
+	 * parameter.
+	 * 
+	 * @param textPane
+	 *            the text pane
+	 * @return the number of lines needed for displaying the text contained in
+	 *         the text pane.
+	 */
+	private int countTextLines(final JTextPane textPane) {
+
+		int lineCount = 0;
+		final int totChars = textPane.getText().length();
+		if (totChars > 0) {
+			try {
+				int offset = totChars;
+				while (offset > 0) {
+					offset = Utilities.getRowStart(textPane, offset) - 1;
+					lineCount++;
+				}
+			} catch (BadLocationException e) {
+				Logger.getLogger(SegmentView.class)
+						.warn("countTextLines method - Error occurred while retrieving the lines number for displaying the text",
+								e);
```

Comment:
The SegmentView class already has a Logger instance called `LOG`, so you can just use that and don't need to called `Logger.getLogger` again.